reana_server/rest/workflows.py

```diff
@@ -1154,8 +1161,32 @@ def upload_file(workflow_id_or_name, user):  # noqa
                 "message": "Internal server error."
               }
     """
+
+    def _prevent_disk_quota_excess(user, file_bytes, filename):
+        """
+        Prevent potential disk quota excess.
+
+        E.g. when uploading big files.
+        """
+        disk_resource = get_default_quota_resource(ResourceType.disk.name)
+        user_resource = UserResource.query.filter_by(
+            user_id=user.id_, resource_id=disk_resource.id_
+        ).first()
+        if (
+            user_resource.quota_limit > 0
+            and user_resource.quota_used + file_bytes > user_resource.quota_limit
+        ):
+            human_readable_limit = ResourceUnit.human_readable_unit(
+                ResourceUnit.bytes_, user_resource.quota_limit
+            )
+            raise REANAQuotaExceededError(
```

Comment:
Spotted while testing this PR group, but not introduced in it (I've tested and the R-Server reply is correct, so error happens in client).

With the following setup:
```console
$ reana-dev git-status
reana-ui @ pr-154 @ 42a7180 profile: hide quotas and gitlab when no token
reana-workflow-engine-yadage @ master @ 0cb641b Merge branch 'maint-0.7'
reana-workflow-engine-serial @ master @ 786524a Merge branch 'maint-0.7'
reana-workflow-engine-cwl @ master @ 88cf46c Merge branch 'maint-0.7'
reana-message-broker @ master @ 6fdfe12 Merge branch 'maint-0.7'
pytest-reana @ master @ 8315bc7 Merge branch 'maint-0.7'
reana-workflow-controller @ pr-349 @ 3a6c403 api: recalculate quota on file upload/deletion
reana-auth-vomsproxy @ master @ a315022 release: v1.0.0
reana-auth-krb5 @ master @ 29931aa ci: added github actions workflow
reana-commons @ master @ 737a2f6 Merge branch 'maint-0.7'
reana-db @ pr-112 @ ccd1e00 utils: enrich disk quota calculation functions
reana-server @ pr-317 @ 73dc0b6 api: prevent disk quota excess on file upload
reana-job-controller @ master @ fb4e363 Merge branch 'maint-0.7'
$ reana-client quota-show --resource disk
190.43 MiB out of 190.73 MiB used (100%)
$ ls -lah *-file.root
-rw-r--r--  1 rodrigdi  staff    24M Nov 16 10:34 medium-file.root
-rw-r--r--  1 rodrigdi  staff   7.3M Nov 16 10:32 small-file.root
```

- Scenario I, uploading `small-file.root` shows the correct message, uploading `medium-file.root` doesn't:
```console
$ reana-client upload small-file.root -w workflow
Something went wrong while uploading /Users/rodrigdi/project/reana/src/reana-demo-helloworld/small-file.root: 
Uploading file /small-file.root would exceed the disk quota limit (190.73 MiB). Aborting.
$ reana-client upload medium-file.root -w workflow
Something went wrong while uploading /Users/rodrigdi/project/reana/src/reana-demo-helloworld/medium-file.root: 
Could not connect to the server https://localhost
```

- Scenario II, retrying quickly shows a weird error (probably hard to happen in real life but still):
```console
$ reana-client upload small-file.root -w workflow
Something went wrong while uploading /Users/rodrigdi/project/reana/src/reana-demo-helloworld/small-file.root: 
Uploading file /small-file.root would exceed the disk quota limit (190.73 MiB). Aborting.
$ reana-client upload small-file.root -w workflow
Something went wrong while uploading /Users/rodrigdi/project/reana/src/reana-demo-helloworld/small-file.root: 
Expecting value: line 1 column 1 (char 0)
```

@tiborsimko did you spot something similar when testing this?

Reply:
I can reproduce it. It looks like a problem when uploading _relatively_ big files. [This exception](https://github.com/reanahub/reana-client/blob/d2a5f2ffe83f5795da9e0cdcb3657d2157893a61/reana_client/api/client.py#L322-L323) gets triggered in the client whose content is: 

```python
ConnectionError(ProtocolError('Connection aborted.', BrokenPipeError(32, 'Broken pipe')))
```

My humble suspicion is that we cut the "stream" connection before it's completed thus resulting in a broken pipe, but I might be wrong. Probably you have better knowledge on this @diegodelemos. Is there any way to gracefully cut the stream connecion?